packages/engine/src/raycast/classes/Raycast.ts

```diff
@@ -0,0 +1,102 @@
+import { Camera, Object3D, Raycaster, Renderer, Vector2 } from "three";
+
+export class RayCast extends Object3D{
+
+    rayCaster : Raycaster;
+    // vrControl: VRControl;
+    mouse:Vector2;
+    selectState:Boolean;
+    objsToTest: [];
+
+    constructor(){
+        super();
+        this.init();
+    }
+
+    addObject(object){
+        this.objsToTest.push(object);
+    }
+
+    init(){
+        this.rayCaster = new Raycaster();
+        this.selectState = false;
+
+        this.mouse = new Vector2();
+        this.mouse.x = this.mouse.y = null;
+
+        window.addEventListener( 'pointermove', ( event )=>{
```

Comment:
window events are problematic for if we need to use the raycasting system on the server. this should be integrating with the RaycastSystem, ClientInputSystem and ActionSystem instead of a class that exists on components.

Reply:
this also doesn't work cross-platform